lib/api/dsl/storage/removeOperands.js

```diff
@@ -167,11 +167,15 @@ class OperandsRemoval {
       field = Object.keys(condition.value)[0];
 
     if (operand.fields[field].count > 1) {
-      const info = operand.fields[field].subfilters[subfilter.id];
+      const info = operand.fields[field].subfilters[subfilter.id][condition.id];
 
       operand.fields[field].tree.remove(info.low, info.high, info.subfilter);
       operand.fields[field].count--;
-      delete operand.fields[field].subfilters[subfilter.id];
+      delete operand.fields[field].subfilters[subfilter.id][condition.id];
+
+      if (Object.keys(operand.fields[field].subfilters[subfilter.id]).length === 0) {
+        delete operand.fields[field].subfilters[subfilter.id];
+      }
```

Comment:
There is an unnecessary deletion occuring if there is exactly 1 condition left in the structure: the condition deletion, and the subfilter one.

This can be prevented with the following code, replacing the lines 174-178:

```js
    if (Object.keys(operand.fields[field].subfilters[subfilter.id]).length === 1) {
      delete operand.fields[field].subfilters[subfilter.id];
    }
    else {
      delete operand.fields[field].subfilters[subfilter.id][condition.id];
    }
```